src/wmtk/multimesh/utils/extract_child_mesh_from_tag.cpp

```diff
@@ -114,7 +114,53 @@ std::shared_ptr<Mesh> internal::TupleTag::extract_and_register_child_mesh_from_t
         m.register_child_mesh(child_ptr, child_to_parent_map);
         return child_ptr;
     }
-    case PrimitiveType::Tetrahedron: throw("not implemented");
+    case PrimitiveType::Tetrahedron: {
+        std::map<int64_t, int64_t> parent_to_child_vertex_map;
+        int64_t child_vertex_count = 0;
+
+        RowVectors4l tet_mesh_matrix;
+
+        tet_mesh_matrix.resize(tagged_tuples.size(), 4);
+        for (int64_t i = 0; i < tagged_tuples.size(); ++i) {
+            const std::array<int64_t, 4> vs = {
+                {m.id(tagged_tuples[i], PrimitiveType::Vertex),
+                 m.id(m.switch_vertex(tagged_tuples[i]), PrimitiveType::Vertex),
+                 m.id(
+                     m.switch_vertex(m.switch_edge(m.switch_face(tagged_tuples[i]))),
+                     PrimitiveType::Vertex),
+                 m.id(m.switch_vertex(m.switch_edge(tagged_tuples[i])), PrimitiveType::Vertex)}};
+
+            // std::cout << vs[0] << " " << vs[1] << " " << vs[2] << " " << vs[3] << " " <<
+            // std::endl;
+
+            for (int k = 0; k < 4; ++k) {
+                size_t size = parent_to_child_vertex_map.size();
+                parent_to_child_vertex_map.try_emplace(vs[k], size);
+                tet_mesh_matrix(i, k) = parent_to_child_vertex_map[vs[k]];
+            }
+        }
+
+        // for (auto key : parent_to_child_vertex_map) {
+        //     std::cout << key.first << " --> " << key.second << std::endl;
+        // }
+
+        // std::cout << tet_mesh_matrix << std::endl;
+
+        std::shared_ptr<TetMesh> child_ptr = std::make_shared<TetMesh>();
+        auto& child = *child_ptr;
+        child.initialize(tet_mesh_matrix);
```

Comment:
ok, this is following what was done previously and lets leave it as this for now, but in the future we really shouldn't need indices here lol. the EV / FE / TF matrices should suffice for generating meshes lol